lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,10 +71,59 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    notify_status if data['rebootrequired'] == true
+
+    data
   end
 
+  def notify_status
+    Puppet.warning "A reboot is required to progress further. Notifying Puppet."
```

Comment:
I think we ideally want 2 messages here... one that we were able to request the reboot.

Another when we couldn't find a matching resource in the graph to assist with triggering the reboot